### PR TITLE
Replacing `BOOL` with `bool`

### LINKED ISSUE #48317: Use standard booleans throughout RyuJit
Some methods in the compiler are still using `BOOL`. This leads to amusing situations like in this method:
https://github.com/dotnet/runtime/blob/56d0eb94ec83473d99c1050072a224da327bc02a/src/coreclr/jit/importer.cpp#L18922-L18952

There is some value in using `bool` everywhere. This is not a high priority item, logging it so that it doesn't get lost.

This has already been fixed for the Jit/EE interface in #45630.

### PR DESCRIPTION
**Pull request by WhiteBlackGoose**
Replaced all Windows' BOOL to bool in RyuJIT except
for WIN API related cases.

Fix #48317

### CODE REVIEW

File: src/coreclr/jit/lclvars.cpp [hunk 1]
```diff
@@ -564,8 +564,7 @@ void Compiler::lvaInitRetBuffArg(InitVarDscInfo* varDscInfo, bool useFixedRetBuf
             info.compCompHnd->getMethodSig(info.compMethodHnd, &sigInfo);
             assert(JITtype2varType(sigInfo.retType) == info.compRetType); // Else shouldn't have a ret buff.
 
-            info.compRetBuffDefStack =
-                (info.compCompHnd->isStructRequiringStackAllocRetBuf(sigInfo.retTypeClass));
+            info.compRetBuffDefStack = (info.compCompHnd->isStructRequiringStackAllocRetBuf(sigInfo.retTypeClass));
             if (info.compRetBuffDefStack)
```
Comment (WhiteBlackGoose) [inline src/coreclr/jit/lclvars.cpp:568]: Wait, my changes didn't affect this one (it was the suggested patch). CI didn't pass in the main branch, did it? 🦆

**Comment by hoyosjs** [comment]
@WhiteBlackGoose No, it was a break in the CI orchestrator. I've retriggered the build.

**Review by AraHaan — APPROVED** [review]
Looks good (if only constexpr was used though for some of these that only returns booleans and safe checks). I know a lot of things in the stl that uses constexpr and ``noexcept`` for things like those on these headers and things known within the functions callstacks to never throw.

As such I think a lot of the runtime that has bits that never throw should mark ``noexcept`` and ``constexpr`` if possible.

**Comment by WhiteBlackGoose** [comment]
@JulieLeeMSFT could you review, please? I'm not sure if I have to ping someone, but no reaction for this and #51758 PR came from members, so I'm a bit confused 😕

**Comment by JulieLeeMSFT** [comment]
@sandreenko PTAL.

**Comment by JulieLeeMSFT** [comment]
> @JulieLeeMSFT could you review, please? I'm not sure if I have to ping someone, but no reaction for this and #51758 PR came from members, so I'm a bit confused 😕

@WhiteBlackGoose sorry for the delay. Pinged @sandreenko for code review.

**Comment by sandreenko** [comment]
> @JulieLeeMSFT could you review, please? I'm not sure if I have to ping someone, but no reaction for this and #51758 PR came from members, so I'm a bit confused 😕

I recommend you to tag @dotnet/jit-contrib in jit changes when they are ready for review so we get a notification about them. Otherwise, we scroll through all-new PRs from time to time but it would take longer until we notice it.

File: src/coreclr/jit/lclvars.cpp (hunk 1 above)
Comment (sandreenko) [reply src/coreclr/jit/lclvars.cpp:568]: what do you mean? I do not think that the jit-format job deleted `== TRUE);` here.

**Review by sandreenko — APPROVED** [review]
LGTM, thanks @whiteblackgoose!

File: src/coreclr/jit/lclvars.cpp (hunk 1 above)
Comment (WhiteBlackGoose) [reply src/coreclr/jit/lclvars.cpp:568]: Ohhh, right, sorry, I'm blind XD. Yes, it's completely alright.

**Comment by SingleAccretion** [comment]
Red CI seems to be reporting failures, the tests themselves passed.
```
Libraries Test Run checked coreclr Linux x64 Debug
   System.Runtime.Tests  Total: 38821, Errors: 0, Failed: 0, Skipped: 12, Time: 123.107s

Libraries Test Run checked coreclr Linux_musl arm64 Release
   System.IO.FileSystem.Tests  Total: 5519, Errors: 0, Failed: 0, Skipped: 16, Time: 113.020s

Libraries Test Run checked coreclr Linux_musl x64 Debug
   System.Xml.RW.ReaderSettings.Tests  Total: 926, Errors: 0, Failed: 0, Skipped: 0, Time: 2.649s

Worker 1: got error: Traceback (most recent call last):
  File "/datadisks/disk1/work/B71D09DC/p/reporter/run.py", line 44, in run
    self.__process(item)
  File "/datadisks/disk1/work/B71D09DC/p/reporter/run.py", line 34, in __process
    self.publisher.upload_batch(batch)
  File "/datadisks/disk1/work/B71D09DC/p/reporter/azure_devops_result_publisher.py", line 35, in upload_batch
    self.publish_results(test_case_results, test_name_order, results_with_attachments)
  File "/datadisks/disk1/work/B71D09DC/p/reporter/azure_devops_result_publisher.py", line 70, in publish_results
    published_results = test_client.add_test_results_to_test_run(list(test_case_results), self.team_project, self.test_run_id)  # type: List[TestCaseResult]
  File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/v5_1/test/test_client.py", line 679, in add_test_results_to_test_run
    content=content)
  File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 104, in _send
    response = self._send_request(request=request, headers=headers, content=content, media_type=media_type)
  File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 68, in _send_request
    self._handle_error(request, response)
  File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 272, in _handle_error
    status_code=response.status_code))
azure.devops.exceptions.AzureDevOpsClientRequestError: Operation returned an invalid status code of 503.
```

**Comment by AraHaan** [comment]
> Red CI seems to be reporting failures, the tests themselves passed.
> 
> ```
> Libraries Test Run checked coreclr Linux x64 Debug
>    System.Runtime.Tests  Total: 38821, Errors: 0, Failed: 0, Skipped: 12, Time: 123.107s
> 
> Libraries Test Run checked coreclr Linux_musl arm64 Release
>    System.IO.FileSystem.Tests  Total: 5519, Errors: 0, Failed: 0, Skipped: 16, Time: 113.020s
> 
> Libraries Test Run checked coreclr Linux_musl x64 Debug
>    System.Xml.RW.ReaderSettings.Tests  Total: 926, Errors: 0, Failed: 0, Skipped: 0, Time: 2.649s
> 
> Worker 1: got error: Traceback (most recent call last):
>   File "/datadisks/disk1/work/B71D09DC/p/reporter/run.py", line 44, in run
>     self.__process(item)
>   File "/datadisks/disk1/work/B71D09DC/p/reporter/run.py", line 34, in __process
>     self.publisher.upload_batch(batch)
>   File "/datadisks/disk1/work/B71D09DC/p/reporter/azure_devops_result_publisher.py", line 35, in upload_batch
>     self.publish_results(test_case_results, test_name_order, results_with_attachments)
>   File "/datadisks/disk1/work/B71D09DC/p/reporter/azure_devops_result_publisher.py", line 70, in publish_results
>     published_results = test_client.add_test_results_to_test_run(list(test_case_results), self.team_project, self.test_run_id)  # type: List[TestCaseResult]
>   File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/v5_1/test/test_client.py", line 679, in add_test_results_to_test_run
>     content=content)
>   File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 104, in _send
>     response = self._send_request(request=request, headers=headers, content=content, media_type=media_type)
>   File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 68, in _send_request
>     self._handle_error(request, response)
>   File "/home/helixbot/.azdo-env/lib/python3.6/site-packages/azure/devops/client.py", line 272, in _handle_error
>     status_code=response.status_code))
> azure.devops.exceptions.AzureDevOpsClientRequestError: Operation returned an invalid status code of 503.
> ```

503 is ``Unavailable`` meaning DevOps was offline for some reason. Possibly due to an outage or maintenance.

**Comment by hoyosjs** [comment]
Checked on the failures. There's no failed tests. @sandreenko, if you feel like this is ready to merge please do so.

@MattGal I've seen this issue with 503s starting to cause considerable pain, where we'll fail due to reporting even though all tests passed. Helix log searching in runfo is not very easy for this type of failures and Kusto ingestion happens from AzDO afaik so we can't use it. Is there any tracking issue for this?
cc: @aik-jahoda @ericstj @dotnet/dnceng @dotnet/runtime-infrastructure

**Comment by safern** [comment]
@hoyosjs this is the issue tracking that failure: https://github.com/dotnet/core-eng/issues/13026 and they are working on adding retries to the reporter: https://github.com/dotnet/arcade/issues/7371 so maybe consider pinging there to raise their attention that we are being badly impacted?

**Comment by MattGal** [comment]
> Checked on the failures. There's no failed tests. @sandreenko, if you feel like this is ready to merge please do so.
> 
> @MattGal I've seen this issue with 503s starting to cause considerable pain, where we'll fail due to reporting even though all tests passed. Helix log searching in runfo is not very easy for this type of failures and Kusto ingestion happens from AzDO afaik so we can't use it. Is there any tracking issue for this?
> cc: @aik-jahoda @ericstj @dotnet/dnceng @dotnet/runtime-infrastructure

@hoyosjs nothing's changed about how often the 503s occur, they just now actually report failure starting 4/29.  It's a long story, hit me up on Teams for a longer discussion.

https://github.com/dotnet/arcade/issues/7371 is the FR issue that tracks introducing more retries to the script in the case of Azure Devops failure.

**Comment by hoyosjs** [comment]
Wasn't aware they were getting dropped on the floor. AS always, thanks @MattGal :) And thanks @safern.

**Comment by JulieLeeMSFT** [comment]
Thanks @sandreenko for merging it.

**Pull request merged**